chains/evm/contracts/poolsV2/TokenPool.sol

```diff
@@ -42,11 +90,193 @@ abstract contract TokenPool is IPoolV2, TokenPoolV1 {
+  /// @param releaseOrMintIn The input to validate. The remote chain, pool, and token must all be configured.
+  /// @param localAmount The amount to release or mint on the local chain after any decimal conversion.
+  /// @param finality The finality depth requested by the message. A value of zero uses the standard lane.
+  function _validateReleaseOrMint(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn,
+    uint256 localAmount,
+    uint16 finality
+  ) internal {
+    if (!isSupportedToken(releaseOrMintIn.localToken)) revert InvalidToken(releaseOrMintIn.localToken);
+    if (IRMN(i_rmnProxy).isCursed(bytes16(uint128(releaseOrMintIn.remoteChainSelector)))) revert CursedByRMN();
+    _onlyOffRamp(releaseOrMintIn.remoteChainSelector);
+
+    // Validates that the source pool address is configured on this pool.
+    if (!isRemotePool(releaseOrMintIn.remoteChainSelector, releaseOrMintIn.sourcePoolAddress)) {
+      revert InvalidSourcePoolAddress(releaseOrMintIn.sourcePoolAddress);
+    }
+
+    FastFinalityConfig storage finalityConfig = s_finalityConfig;
+    if (finality != 0) {
+      finalityConfig.inboundRateLimiterConfig[releaseOrMintIn.remoteChainSelector]._consume(
+        localAmount, releaseOrMintIn.localToken
+      );
+      emit FastTransferInboundRateLimitConsumed(
+        releaseOrMintIn.remoteChainSelector, releaseOrMintIn.localToken, localAmount
+      );
+    } else {
+      _consumeInboundRateLimit(releaseOrMintIn.remoteChainSelector, localAmount);
+    }
+  }
+
+  // ========================================================
+  // │                          Finality                     │
+  // ==============================================+=========
+
+  /// @notice Updates the finality configuration for token transfers.
+  function applyFinalityConfigUpdates(
+    uint16 finalityThreshold,
+    uint16 fastTransferFeeBps,
+    uint256 maxAmountPerRequest,
+    FastFinalityRateLimitConfigArgs[] calldata rateLimitConfigArgs
+  ) external virtual onlyOwner {
+    FastFinalityConfig storage finalityConfig = s_finalityConfig;
+    finalityConfig.finalityThreshold = finalityThreshold;
+    if (fastTransferFeeBps >= BPS_DIVIDER) {
+      revert InvalidFastTransferFeeBps();
+    }
+    finalityConfig.fastTransferFeeBps = fastTransferFeeBps;
+    finalityConfig.maxAmountPerRequest = maxAmountPerRequest;
+    _setFastFinalityRateLimitConfig(rateLimitConfigArgs);
+    emit FinalityConfigUpdated(finalityThreshold, fastTransferFeeBps, maxAmountPerRequest);
+  }
+
+  /// @notice Sets the fast finality based rate limit configurations for specified remote chains.
+  /// @param rateLimitConfigArgs Array of structs containing remote chain selectors and their rate limiter configs.
+  function setFastFinalityRateLimitConfig(
+    FastFinalityRateLimitConfigArgs[] calldata rateLimitConfigArgs
+  ) external virtual onlyOwner {
+    _setFastFinalityRateLimitConfig(rateLimitConfigArgs);
+  }
+
+  function _setFastFinalityRateLimitConfig(
```

Comment:
Is this pretty much the same code as the other RL setting? Can probably dedup some and make one consistent way to set them